recipes-kernel/linux-firmware/linux-firmware_%.bbappend

```diff
@@ -0,0 +1,75 @@
+# Augments upstream linux-firmware with additional and updated images
+# from Raspbian:
+# https://github.com/RPi-Distro/firmware-nonfree
+# https://github.com/RPi-Distro/bluez-firmware
+
+LICENSE_append = "\
+    & Firmware-cypress \
+"
+
+LIC_FILES_CHKSUM_append = "\
+    file://LICENCE.cypress;md5=cbc5f665d04f741f1e006d2096236ba7 \
+"
+NO_GENERIC_LICENSE[Firmware-cypress] = "LICENCE.cypress"
+
+SRC_URI_append = " \
+    git://github.com/RPi-Distro/firmware-nonfree;destsuffix=raspbian-nf;name=raspbian-nf \
+    git://github.com/RPi-Distro/bluez-firmware;destsuffix=raspbian-bluez;name=raspbian-bluez \
+"
+
+SRCREV_raspbian-nf = "86e88fbf0345da49555d0ec34c80b4fbae7d0cd3"
+SRCREV_raspbian-bluez = "e28cd7ee8615de33aa7ec2b41d556af61a4a2707"
+SRCREV_FORMAT = "git"
+
+do_install_append() {
+    install -d ${D}${nonarch_base_libdir}/firmware/brcm/
+
+    # Replace outdated upstream files with fresh from raspbian
+    # firmware-nonfree.  Raspbian includes blobs and nvram definitions
+    # that are also necessary.
+    for fw in brcmfmac43430-sdio brcmfmac43455-sdio ; do
+      install -m 0644 ${WORKDIR}/raspbian-nf/brcm/${fw}.* ${D}${nonarch_base_libdir}/firmware/brcm/
+    done
+
+    # Add missing Cypress Bluetooth files from raspbian bluez-firmware
+    for fw in BCM43430A1.hcd BCM4345C0.hcd ; do
+        install -m 0644 ${WORKDIR}/raspbian-bluez/broadcom/${fw} ${D}${nonarch_base_libdir}/firmware/brcm/
+    done
+}
+
+PACKAGES =+ "\
+    ${PN}-bcm43455 \
+    ${PN}-cypress-license \
+    ${PN}-bcm43430a1-hcd \
+    ${PN}-bcm4345c0-hcd \
+"
+
+# For additional Broadcom
+LICENSE_${PN}-bcm43455 = "Firmware-broadcom_bcm43xx"
+
+FILES_${PN}-bcm43430 += " \
```

Comment:
I remember that in the past version of the recipe, where I provided a PR to include the NVRAM file, I had to use `_append` instead of `+=`, otherwise the build would't include the file. I don't know exactly why.

Check here: https://github.com/hhromic/meta-raspberrypi/blob/fc964659c46b50b69551eadd834d21c81777b00b/recipes-kernel/linux-firmware/linux-firmware_%25.bbappend#L24

Reply:
That was due to the presence of the machine override on the LHS.  Since those are coming back, this will be switched to `_append`.

Reply:
Ah right, the `_rpi` override. I remember now. Thanks for your patience :)

Reply:
No problem; I'm learning as I go....